### PR TITLE
Updating the README

### PR DESCRIPTION
**Pull request by kelset**
This PR is aimed to provide the users (old & new) with a more comprehensive description of many concepts, like:
* What this library is
* How they can contribute
* What they should when facing issues

Aside from that, I think the first step towards a more in-depth PR to the docs is to provide the links @matthamil had separately [here](https://github.com/react-navigation/react-navigation-links) in the README to give who start at least some extra resources "right away".

Moreover, the FAQ section aims to address some recurring issues with the goal of augmenting the transparency from us maintainers to the users.

Yes, this would mean a "longer" README but since it's usually the place many devs look into to "understand" a library I feel it's worth the tradeoff.

Obviously completely open to any feedback.

### CODE REVIEW

**Comment by vonovak** [comment]
there are some 3-rd party packages for react-navigation that might be worth mentioning: I know of [react-navigation-addons](https://github.com/satya164/react-navigation-addons) and [ehm, shameless plug] [react-navigation-props-mapper](https://github.com/vonovak/react-navigation-props-mapper)

**Comment by kelset** [comment]
Hey @vonovak thanks for your feedback, yeah a section with "extra libs" can be a good section. I'll add it tomorrow.

@matthamil already provided me some feedback, still waiting on @spencercarli and/or @GantMan to let me know if it's fine for them.

File: README.md [hunk 1]
```diff
@@ -1,63 +1,138 @@
...
-
+# React Navigation [![CircleCI](https://circleci.com/gh/react-community/react-navigation/tree/master.svg?style=shield&circle-token=622fcb1d78413084c2f44699ed2104246a177485)](https://circleci.com/gh/react-community/react-navigation/tree/master) [![npm version](https://badge.fury.io/js/react-navigation.svg)](https://badge.fury.io/js/react-navigation) [![codecov](https://codecov.io/gh/react-community/react-navigation/branch/master/graph/badge.svg)](https://codecov.io/gh/react-community/react-navigation) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactnavigation.org/docs/guides/contributors)
 
 *Learn once, navigate anywhere.*
 
-Browse the docs on [reactnavigation.org](https://reactnavigation.org/) or try it out on [our expo demo](https://exp.host/@react-navigation/NavigationPlayground).
-
```
Comment (GantMan) [inline README.md]: I'm not sure you should remove the expo demo.   This should be moved at best, but we should keep this demo in the docs.

Comment (kelset) [reply README.md]: Got it. I'll update accordingly.

**Comment by kelset** [comment]
Ok update with the expo link for @GantMan and the libs for @vonovak ;)

I think now it's solid 💯 thanks for the feedback.

**Pull request merged**